pkg/internal/controller/controller_test.go

```diff
@@ -169,6 +172,108 @@ var _ = Describe("controller", func() {
 			close(done)
 		}, 10.0)
 
+		It("should not error when source.channel is passed", func(done Done) {
+			eventCount := 0
+			// channel to be closed when event is processed
+			processed := make(chan struct{})
+			// source channel to be injected
+			ch := make(chan event.GenericEvent, 1)
+
+			ctx, cancel := context.WithCancel(context.TODO())
+			defer cancel()
+
+			// event to be sent to the channel
+			p := &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
+			}
+			evt := event.GenericEvent{
+				Object: p,
+			}
+
+			ins := &source.Channel{Source: ch}
+			ins.DestBufferSize = 1
+			Expect(inject.StopChannelInto(ctx.Done(), ins)).To(BeTrue())
```

Comment:
We just deprecated the injection, lets not add more code that uses it

Reply:
While starting the source (with channel passed) we check if the stop channel has been injected and return an error if not present ([here](https://github.com/kubernetes-sigs/controller-runtime/blob/a8c19c49e49cfba2aa486ff322cbe5222d6da533/pkg/source/source.go#L208)). I think we may have to modify the implementation while [`starting`](https://github.com/kubernetes-sigs/controller-runtime/blob/a8c19c49e49cfba2aa486ff322cbe5222d6da533/pkg/source/source.go#L197) source.channel to remove this or is there any other way to go about this?